// @strapi/strapi/dist/commands/actions/watch-admin/action.js
"use strict";
const strapiAdmin = require("@strapi/admin");
const strapiUtils = require("@strapi/utils");
const getEnabledPlugins = require("../../../core/loaders/plugins/get-enabled-plugins.js");
const addSlash = require("../../../utils/addSlash.js");
const Strapi = require("../../../Strapi.js");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const strapiAdmin__default = /* @__PURE__ */ _interopDefault(strapiAdmin);
const action = async ({ browser }) => {
  const appContext = await Strapi.compile();
  const strapiInstance = Strapi({
    ...appContext,
    autoReload: true,
    serveAdminPanel: false
  });
  const plugins = await getEnabledPlugins.getEnabledPlugins(strapiInstance, { client: true });
  const { adminPath } = strapiUtils.getConfigUrls(strapiInstance.config, true);
  const adminPort = strapiInstance.config.get("admin.port", 8e3);
  const adminHost = strapiInstance.config.get("admin.host", "localhost");
  const backendURL = strapiUtils.getAbsoluteServerUrl(strapiInstance.config, true);
  strapiAdmin__default.default.watchAdmin({
    appDir: appContext.appDir,
    buildDestDir: appContext.distDir,
    plugins,
    port: adminPort,
    host: adminHost,
    browser,
    options: {
      backend: backendURL,
      adminPath: addSlash(adminPath)
    }
  });
};
module.exports = action;
