// @strapi/admin/build/7811.fdbe09af.chunk.js
(self.webpackChunk_strapi_admin=self.webpackChunk_strapi_admin||[]).push([[7811],{56671:function(f,u,n){var r=n(35897),o=n(74741),a=o(r);f.exports=a},52920:function(f,u,n){var r=n(56671),o=n(81580);function a(s,e){var i=-1,t=o(s)?Array(s.length):[];return r(s,function(d,x,h){t[++i]=e(d,x,h)}),t}f.exports=a},50690:function(f,u,n){var r=n(26460),o=n(80166),a=n(88904),s=n(52920),e=n(36024),i=n(76535),t=n(1397),d=n(15362),x=n(77236);function h(E,c,l){c.length?c=r(c,function(p){return x(p)?function(v){return o(v,p.length===1?p[0]:p)}:p}):c=[d];var m=-1;c=r(c,i(a));var g=s(E,function(p,v,j){var P=r(c,function(O){return O(p)});return{criteria:P,index:++m,value:p}});return e(g,function(p,v){return t(p,v,l)})}f.exports=h},36024:function(f){function u(n,r){var o=n.length;for(n.sort(r);o--;)n[o]=n[o].value;return n}f.exports=u},40709:function(f,u,n){var r=n(22945);function o(a,s){if(a!==s){var e=a!==void 0,i=a===null,t=a===a,d=r(a),x=s!==void 0,h=s===null,E=s===s,c=r(s);if(!h&&!c&&!d&&a>s||d&&x&&E&&!h&&!c||i&&x&&E||!e&&E||!t)return 1;if(!i&&!d&&!c&&a<s||c&&e&&t&&!i&&!d||h&&e&&t||!x&&t||!E)return-1}return 0}f.exports=o},1397:function(f,u,n){var r=n(40709);function o(a,s,e){for(var i=-1,t=a.criteria,d=s.criteria,x=t.length,h=e.length;++i<x;){var E=r(t[i],d[i]);if(E){if(i>=h)return E;var c=e[i];return E*(c=="desc"?-1:1)}}return a.index-s.index}f.exports=o},74741:function(f,u,n){var r=n(81580);function o(a,s){return function(e,i){if(e==null)return e;if(!r(e))return a(e,i);for(var t=e.length,d=s?t:-1,x=Object(e);(s?d--:++d<t)&&i(x[d],d,x)!==!1;);return e}}f.exports=o},98978:function(f,u,n){var r=n(40314),o=n(50690),a=n(94569),s=n(63108),e=a(function(i,t){if(i==null)return[];var d=t.length;return d>1&&s(i,t[0],t[1])?t=[]:d>2&&s(t[0],t[1],t[2])&&(t=[t[0]]),o(i,r(t,1),[])});f.exports=e},99226:function(f,u,n){"use strict";n.d(u,{A:function(){return i}});var r=n(74512),o=n(8471),a=n(77981);const s=(0,o.ZP)(a.x)`
  display: grid;
  grid-template-columns: ${({hasSideNav:t})=>t?"auto 1fr":"1fr"};
`,e=(0,o.ZP)(a.x)`
  overflow-x: hidden;
`,i=({sideNav:t,children:d})=>(0,r.jsxs)(s,{hasSideNav:Boolean(t),children:[t,(0,r.jsx)(e,{paddingBottom:10,children:d})]})},84458:function(f,u,n){"use strict";n.d(u,{r:function(){return E}});var r=n(74512),o=n(32735),a=n(95441),s=n(14911),e=n(8471),i=n(59619),t=n(8365),d=n(77981);const x=e.ZP.a`
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  pointer-events: ${({disabled:c})=>c?"none":void 0};
  color: ${({disabled:c,theme:l})=>c?l.colors.neutral600:l.colors.primary600};

  svg path {
    transition: fill 150ms ease-out;
    fill: currentColor;
  }

  svg {
    font-size: ${10/16}rem;
  }

  ${t.Z} {
    transition: color 150ms ease-out;
    color: currentColor;
  }

  &:hover {
    color: ${({theme:c})=>c.colors.primary500};
  }

  &:active {
    color: ${({theme:c})=>c.colors.primary700};
  }

  ${i.BF};
`,h=(0,e.ZP)(d.x)`
  display: flex;
`,E=o.forwardRef(({children:c,href:l,to:m,disabled:g=!1,startIcon:p,endIcon:v,...j},P)=>{const O=l?"_blank":void 0,M=l?"noreferrer noopener":void 0;return(0,r.jsxs)(x,{as:m&&!g?s.OL:"a",target:O,rel:M,to:g?void 0:m,href:g?"#":l,disabled:g,ref:P,...j,children:[p&&(0,r.jsx)(h,{as:"span","aria-hidden":!0,paddingRight:2,children:p}),(0,r.jsx)(t.Z,{children:c}),v&&!l&&(0,r.jsx)(h,{as:"span","aria-hidden":!0,paddingLeft:2,children:v}),l&&(0,r.jsx)(h,{as:"span","aria-hidden":!0,paddingLeft:2,children:(0,r.jsx)(a.Z,{})})]})})},7141:function(f,u,n){"use strict";n.d(u,{m:function(){return i}});var r=n(74512),o=n(8471),a=n(10710);const s=`${232/16}rem`,e=(0,o.ZP)(a.r)`
  width: ${s};
  background: ${({theme:t})=>t.colors.neutral100};
  position: sticky;
  top: 0;
  height: 100vh;
  overflow-y: auto;
  border-right: 1px solid ${({theme:t})=>t.colors.neutral200};
  z-index: 1;
`,i=({ariaLabel:t,...d})=>(0,r.jsx)(e,{"aria-label":t,as:"nav",...d})},65352:function(f,u,n){"use strict";n.d(u,{p:function(){return p}});var r=n(74512),o=n(32735),a=n(17229),s=n(8471),e=n(56374),i=n(37808);const t=v=>{const j=(0,o.useRef)();return(0,o.useEffect)(()=>{j.current=v}),j.current};var d=n(2775),x=n(77981),h=n(56116),E=n(41604),c=n(84642),l=n(8365),m=n(14763);const g=(0,s.ZP)(d.i)`
  width: ${24/16}rem;
  background-color: ${({theme:v})=>v.colors.neutral200};
`,p=({as:v="h2",label:j,searchLabel:P="",searchable:O=!1,onChange:M=()=>{},value:C="",onClear:y=()=>{},onSubmit:D=()=>{},id:A})=>{const[_,T]=(0,o.useState)(!1),W=t(_),K=(0,i.M)(A),L=(0,o.useRef)(void 0),R=(0,o.useRef)(void 0);(0,o.useEffect)(()=>{_&&L.current&&L.current.focus(),W&&!_&&R.current&&R.current.focus()},[_,W]);const $=()=>{T(B=>!B)},U=B=>{y(B),L.current.focus()},Z=B=>{B.relatedTarget?.id!==K&&T(!1)},S=B=>{B.key===e.y.ESCAPE&&T(!1)};return _?(0,r.jsxs)(x.x,{paddingLeft:4,paddingTop:5,paddingBottom:2,paddingRight:4,children:[(0,r.jsx)(h.U,{children:(0,r.jsx)(E.w,{name:"searchbar",value:C,onChange:M,placeholder:"e.g: strapi-plugin-abcd",onKeyDown:S,ref:L,onBlur:Z,onClear:U,onSubmit:D,clearLabel:"Clear",size:"S",children:P})}),(0,r.jsx)(x.x,{paddingLeft:2,paddingTop:4,children:(0,r.jsx)(g,{})})]}):(0,r.jsxs)(x.x,{paddingLeft:6,paddingTop:6,paddingBottom:2,paddingRight:4,children:[(0,r.jsxs)(c.k,{justifyContent:"space-between",alignItems:"flex-start",children:[(0,r.jsx)(l.Z,{variant:"beta",as:v,children:j}),O&&(0,r.jsx)(m.h,{ref:R,onClick:$,label:P,icon:(0,r.jsx)(a.Z,{})})]}),(0,r.jsx)(x.x,{paddingTop:4,children:(0,r.jsx)(g,{})})]})}},89274:function(f,u,n){"use strict";n.d(u,{E:function(){return c}});var r=n(74512),o=n(32735),a=n(48715),s=n(8471),e=n(77981),i=n(8365),t=n(84642),d=n(42561);const x=(0,s.ZP)(e.x)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-decoration: none;
  color: ${({theme:l})=>l.colors.neutral800};
  svg > * {
    fill: ${({theme:l})=>l.colors.neutral600};
  }

  &.active {
    ${({theme:l})=>`
      background-color: ${l.colors.primary100};
      border-right: 2px solid ${l.colors.primary600};
      svg > * {
        fill: ${l.colors.primary700};
      }
      ${i.Z} {
        color: ${l.colors.primary700};
        font-weight: 500;
      }
      `}
  }

  &:focus-visible {
    outline-offset: -2px;
  }
`,h=(0,s.ZP)(a.Z)`
  width: ${12/16}rem;
  height: ${4/16}rem;
  * {
    fill: ${({theme:l,$active:m})=>m?l.colors.primary600:l.colors.neutral600};
  }
`,E=s.ZP.div`
  svg {
    height: ${12/16}rem;
    width: ${12/16}rem;
  }
`,c=o.forwardRef(({children:l,icon:m=null,withBullet:g=!1,as:p=d.f,isSubSectionChild:v=!1,...j},P)=>(0,r.jsxs)(x,{as:p,icon:m,background:"neutral100",paddingLeft:v?9:7,paddingBottom:2,paddingTop:2,ref:P,...j,children:[(0,r.jsxs)(t.k,{children:[m?(0,r.jsx)(E,{children:m}):(0,r.jsx)(h,{}),(0,r.jsx)(e.x,{paddingLeft:2,children:(0,r.jsx)(i.Z,{as:"span",children:l})})]}),g&&(0,r.jsx)(e.x,{as:t.k,paddingRight:4,children:(0,r.jsx)(h,{$active:!0})})]}))},88791:function(f,u,n){"use strict";n.d(u,{D:function(){return m}});var r=n(74512),o=n(32735),a=n(8471),s=n(91917),e=n(84642),i=n(77981),t=n(8365);const d=(0,a.ZP)(e.k)`
  border: none;
  padding: 0;
  background: transparent;
`,x=a.ZP.div`
  display: flex;
  align-items: center;
  transform: rotateX(${({rotated:g})=>g?"0deg":"180deg"});
`,h=({collapsable:g=!1,label:p,onClick:v=()=>{},ariaExpanded:j,ariaControls:P})=>g?(0,r.jsxs)(d,{as:"button",onClick:v,"aria-expanded":j,"aria-controls":P,textAlign:"left",children:[(0,r.jsx)(i.x,{paddingRight:1,children:(0,r.jsx)(t.Z,{variant:"sigma",textColor:"neutral600",children:p})}),g&&(0,r.jsx)(x,{rotated:j,children:(0,r.jsx)(s.Z,{"aria-hidden":!0})})]}):(0,r.jsx)(d,{children:(0,r.jsx)(i.x,{paddingRight:1,children:(0,r.jsx)(t.Z,{variant:"sigma",textColor:"neutral600",children:p})})});var E=n(37808),c=n(14298);const l=(0,a.ZP)(i.x)`
  svg {
    height: ${4/16}rem;
    path {
      fill: ${({theme:g})=>g.colors.neutral500};
    }
  }
`,m=({collapsable:g=!1,label:p,badgeLabel:v,children:j,id:P})=>{const[O,M]=(0,o.useState)(!0),C=(0,E.M)(P),y=()=>{M(D=>!D)};return(0,r.jsxs)(e.k,{direction:"column",alignItems:"stretch",gap:1,children:[(0,r.jsx)(l,{paddingLeft:6,paddingTop:2,paddingBottom:2,paddingRight:4,children:(0,r.jsxs)(i.x,{position:"relative",paddingRight:v?6:0,children:[(0,r.jsx)(h,{onClick:y,ariaExpanded:O,ariaControls:C,collapsable:g,label:p}),v&&(0,r.jsx)(c.C,{backgroundColor:"neutral150",textColor:"neutral600",position:"absolute",right:0,top:"50%",transform:"translateY(-50%)",children:v})]})}),(!g||O)&&(0,r.jsx)("ol",{id:C,children:o.Children.map(j,(D,A)=>(0,r.jsx)("li",{children:D},A))})]})}},36473:function(f,u,n){"use strict";n.d(u,{Z:function(){return e}});var r=n(74512),o=n(32735),a=n(77981),s=n(84642);const e=({children:i,spacing:t=2,horizontal:d=!1,...x})=>(0,r.jsx)(a.x,{paddingTop:2,paddingBottom:4,children:(0,r.jsx)(s.k,{as:"ol",gap:t,direction:d?"row":"column",alignItems:d?"center":"stretch",...x,children:o.Children.map(i,(h,E)=>(0,r.jsx)("li",{children:h},E))})})},48715:function(f,u,n){"use strict";n.d(u,{Z:function(){return a}});var r=n(74512);const o=s=>(0,r.jsx)("svg",{xmlns:"http://www.w3.org/2000/svg",width:"1rem",height:"1rem",fill:"none",viewBox:"0 0 4 4",...s,children:(0,r.jsx)("rect",{width:4,height:4,fill:"#A5A5BA",rx:2})}),a=o},36328:function(f,u,n){"use strict";n.d(u,{Z:function(){return a}});var r=n(74512);const o=s=>(0,r.jsx)("svg",{xmlns:"http://www.w3.org/2000/svg",width:"1rem",height:"1rem",fill:"none",viewBox:"0 0 24 24",...s,children:(0,r.jsx)("path",{fill:"#8E8EA9",d:"M19 10h1a1 1 0 0 1 1 1v10a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V11a1 1 0 0 1 1-1h1V9a7 7 0 0 1 14 0v1Zm-2 0V9A5 5 0 0 0 7 9v1h10Zm-6 4v4h2v-4h-2Z"})}),a=o}}]);
